File: typescript/src/routes/metrics.ts
import express, { Request, Response, Router } from "express";
import os from "os";
import metrics from "../lib/metrics";

const router: Router = express.Router();

router.get("/metrics", (req: Request, res: Response) => {
  res.json({
    free: os.freemem(),
    total: os.totalmem(),
  });
});

export default router;

File: python/src/nucleoid/routes/metrics.py
"""
Metrics route - provides system metrics and resource usage information.
"""
from flask import Blueprint, jsonify, request
import psutil
from typing import Any


metrics_bp = Blueprint('metrics', __name__)


@metrics_bp.route('/metrics', methods=['GET'])
def get_metrics() -> Any:
    """
    Get system metrics including memory usage.

    Returns:
        JSON response with free and total memory
    """
    memory = psutil.virtual_memory()

    return jsonify({
        'free': memory.available,
        'total': memory.total,
    })
